helm/aws-load-balancer-controller/templates/deployment.yaml

```diff
@@ -27,7 +27,7 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
+        {{- if not .Values.serviceMonitor.enabled and .Values.prometheusScrape.enabled}}
```

Comment:
Wrong usage of helm `and` keyword.
Correct is
{{- if and (not .Values.serviceMonitor.enabled) .Values.prometheusScrape.enabled }}

Reply:
Thanks good catch